src/emqx_client.erl

```diff
@@ -562,15 +628,27 @@ mqtt_connect(State = #state{client_id   = ClientId,
 waiting_for_connack(cast, ?CONNACK_PACKET(?RC_SUCCESS,
                                           SessPresent,
                                           Properties),
-                    State = #state{properties = AllProps}) ->
+                    State = #state{properties = AllProps,
+                                   client_id = ClientId}) ->
     case take_call(connect, State) of
         {value, #call{from = From}, State1} ->
             AllProps1 = case Properties of
                             undefined -> AllProps;
                             _ -> maps:merge(AllProps, Properties)
                         end,
             Reply = {ok, self(), Properties},
-            State2 = State1#state{properties = AllProps1,
+            AssignedClientId = fun(<<>>, Props) ->
+                                   case maps:find('Assigned-Client-Identifier', Props) of
+                                       {ok, Value} ->
+                                           Value;
+                                       _ ->
+                                           emqx_logger:error("Bad Client Id")
```

Comment:
`AssignedClientId` will be bound with `ok` atom

Reply:
if it is meant to crash, call `erlang:error(bad_client_id)`
otherwise use a sane default value ?